model.py

```diff
@@ -106,15 +82,21 @@ def inference(self, device):
         with torch.no_grad():
             test_y_pred = self.net(utils.test_X.float().cuda(device)).squeeze()
             test_loss = mse(utils.test_Y.float().cuda(device), test_y_pred).detach().item()
-        print("Testing Loss: " + str("%.5f" % test_loss))
+        print("Testing Loss: " + str("%.5f" % test_loss) + " MAPE: " + str(mape(utils.test_Y.float(), test_y_pred.cpu())) )
         nbins = 100
-        plt.hist(utils.test_Y.numpy(), bins=nbins, histtype = "step", color = "r", label = "Test Dataset")
-        plt.hist(test_y_pred.detach().cpu().numpy(), bins=nbins, histtype = "step", label = "DNN Prediction")
-        plt.xlabel("-log_10(DY weight)")
-        plt.ylabel("events")
-        plt.legend()
+        # plt.hist(utils.test_Y.numpy(), bins=nbins, histtype = "step", color = "r", label = "Test Dataset")
+        # plt.hist(test_y_pred.detach().cpu().numpy(), bins=nbins, histtype = "step", label = "DNN Prediction")
+        # plt.xlabel("-log_10(DY weight)")
+        # plt.ylabel("events")
+        # plt.legend()
+        # plt.savefig(self.manager.args["save_testing_histogram_at"])
+        hist_1 = hist.Hist(hist.axis.Regular(nbins, 0, 10, name="-log_10(DY weight)")).fill(utils.test_Y.numpy())
+        hist_2 = hist.Hist(hist.axis.Regular(nbins, 0, 10, name="-log_10(DY weight)")).fill(test_y_pred.detach().cpu().numpy())
+        fig = plt.figure(figsize=(10, 8))
+        main_ax_artists, sublot_ax_arists = hist_1.plot_ratio(hist_2, rp_ylabel=r"Ratio", rp_num_label="Test Dataset", rp_denom_label="DNN Prediction", rp_uncert_draw_type="line")
+        plt.yticks(np.arange(-5,6))
+        plt.grid()
```

Comment:
This is fine for now, but we can go and specify which axis objects we want this applied to later on rather than using the `plt` API.